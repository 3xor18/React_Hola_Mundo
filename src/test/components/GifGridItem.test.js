import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Prueba de GidGrid', () => {
	const title = 'Titulo';
	const url = 'Url';
	const wrapper = shallow(<GifGridItem title={title} url={url} />);

	test('Debe Generar el componente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('titulo en el parrafo', () => {
		const p = wrapper.find('p');
		expect(p.text().trim()).toBe(title);
	});

	test('image debe contener title y url', () => {
		const img = wrapper.find('img');
		expect(img.prop('src')).toBe(url);
		expect(img.prop('alt')).toBe(title);
	});

	test('Debe contener la clase', () => {
		const clasesEsperadas = 'card animate__animated animate__bounce';
		const div = wrapper.find('div');
		const className = div.prop('className');
		expect(className).toBe(clasesEsperadas);
	});
});
