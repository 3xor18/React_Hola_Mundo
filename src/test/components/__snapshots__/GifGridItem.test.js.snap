// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GidGrid Debe Generar el componente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Titulo"
    src="Url"
  />
  <p>
    Titulo
  </p>
</div>
`;
